File: app/src/main/java/keelim/ArduController/MainActivity.java
package keelim.ArduController;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.os.Message;
import android.support.v7.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    private BluetoothService btService = null;
    private final Handler mHandler = new Handler(){

        @Override
        public void handleMessage(Message msg) {
            super.handleMessage(msg);
        }
    };



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        if(btService == null){
            btService = new BluetoothService(this, mHandler);
        }
    }

    public void onActivityResult(int requestCode, int resultCode, Intent data){
        switch (requestCode){
            case 99:
                if(resultCode == Activity.RESULT_OK){

                } else{

                }
                break;
        }

    }





}

File: app/src/main/java/com/keelim/hardware/view/MainActivity.kt
package com.keelim.hardware.view

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.viewpager.widget.ViewPager
import com.google.android.material.tabs.TabLayout
import com.keelim.hardware.R
import com.keelim.hardware.model.PagerAdapter


import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private lateinit var adapter: PagerAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        tabLayout.apply {
            addTab(tabLayout.newTab().setText(""))
//            addTab(tabLayout.newTab().setText(""))
//            addTab(tabLayout.newTab().setText(""))

            tabGravity = TabLayout.GRAVITY_FILL
        }

        adapter = PagerAdapter(supportFragmentManager)
        viewpager.adapter = adapter
        viewpager.addOnPageChangeListener(object : ViewPager.OnPageChangeListener {
            override fun onPageScrollStateChanged(state: Int) {
                TODO("Not yet implemented")
            }

            override fun onPageScrolled(
                position: Int,
                positionOffset: Float,
                positionOffsetPixels: Int
            ) {
                TODO("Not yet implemented")
            }

            override fun onPageSelected(position: Int) {
                TODO("Not yet implemented")
            }

        })

    }
}